ddp-workspace/projects/ddp-sdk/src/lib/components/internationalization/languageSelector.component.ts

```diff
@@ -15,248 +15,252 @@ import { UserProfileServiceAgent } from '../../services/serviceAgents/userProfil
+            );
     }
-  }
 
-  private getCurrentStoredLangObservable(): Observable<StudyLanguage> {
-    return new Observable<StudyLanguage>(subscriber => {
-      // Use the current language if it exists
-      if (this.currentLanguage) {
-        subscriber.next(this.currentLanguage);
-      } else {
-        // Check for a stored language
-        const loadedCode: string = this.language.useStoredLanguage();
-        if (loadedCode) {
-          const lang: StudyLanguage = this.studyLanguages.find(studyLang => studyLang.languageCode === loadedCode);
-          subscriber.next(lang || null);
+    private getNextObservable(lang: StudyLanguage, obs: Observable<StudyLanguage>): Observable<StudyLanguage> {
+        if (this.foundLanguage(lang)) {
+            return of(lang);
         } else {
-          subscriber.next(null);
+            return obs;
         }
-      }
-      subscriber.complete();
-    });
-  }
+    }
 
-  private getProfileLangObservable(): Observable<StudyLanguage> {
-    const profileLangObservable: Observable<StudyLanguage> = this.profileServiceAgent.profile
-      .pipe(map(profileDecorator => {
-        if (profileDecorator && profileDecorator.profile.preferredLanguage) {
-          return this.studyLanguages.find(studyLang => studyLang.languageCode === profileDecorator.profile.preferredLanguage);
-        } else {
-          return null;
-        }
-      }));
-    return iif(() => this.session.isAuthenticatedSession(), profileLangObservable, of(null));
-  }
+    private getCurrentStoredLangObservable(): Observable<StudyLanguage> {
+        return new Observable<StudyLanguage>(subscriber => {
+            // Use the current language if it exists
+            if (this.currentLanguage) {
+                subscriber.next(this.currentLanguage);
+            } else {
+                // Check for a stored language
+                const loadedCode: string = this.language.useStoredLanguage();
+                if (loadedCode) {
+                    const lang: StudyLanguage = this.studyLanguages.find(studyLang => studyLang.languageCode === loadedCode);
+                    subscriber.next(lang || null);
+                } else {
+                    subscriber.next(null);
+                }
+            }
+            subscriber.complete();
+        });
+    }
+
+    private getProfileLangObservable(): Observable<StudyLanguage> {
+        const profileLangObservable: Observable<StudyLanguage> = this.profileServiceAgent.profile
+            .pipe(map(profileDecorator => {
+                if (profileDecorator && profileDecorator.profile.preferredLanguage) {
+                    return this.studyLanguages.find(studyLang => studyLang.languageCode === profileDecorator.profile.preferredLanguage);
+                } else {
+                    return null;
+                }
+            }));
+        return iif(() => this.hasUserProfile(), profileLangObservable, of(null));
+    }
+
+    private hasUserProfile(): boolean {
```

Comment:
the only change in the file is that I created this function and replaced all places where `this.session.isAuthenticatedSession()` was used